ztools/ztls/handshake_client.go

```diff
@@ -90,10 +90,14 @@ func (c *Conn) clientHandshake() error {
 		}
 	}
 
-	_, err := io.ReadFull(c.config.rand(), hello.random)
-	if err != nil {
-		c.sendAlert(alertInternalError)
-		return errors.New("tls: short read from Rand: " + err.Error())
+	if len(c.config.ClientRandom) == 32 {
+		hello.random = c.config.ClientRandom
```

Comment:
You probably want to copy the random out of the config, rather than just point at it, but I'm not totally convinced either way.
```
copy(hello.random, c.config.ClientRandom)
```